app/controllers/admin/volunteers.py

```diff
@@ -25,81 +23,71 @@ def getVolunteers(query):
         abort(404)
-    eventData = model_to_dict(event, recurse=False)
-    
-    eventData["program"] = event.program
-    trainedParticipantsList = trainedParticipants(event.program, event.term)
-    eventParticipants = getEventParticipants(event)
 
-    isProgramManager = g.current_user.isProgramManagerForEvent(event)
-    bannedUsers = [row.user for row in getBannedUsers(event.program)]
-    if not (g.current_user.isCeltsAdmin or (g.current_user.isCeltsStudentStaff and isProgramManager)):
-        abort(403)
+    # ------------ POST request ------------
+    if request.method == "POST":
+        volunteerUpdated = updateEventParticipants(request.form)
 
-    eventParticipantData = list(EventParticipant.select(EventParticipant, User).join(User).where(EventParticipant.event==event))
-    eventRsvpData = list(EventRsvp.select(EventRsvp, User).join(User).where(EventRsvp.event==event).order_by(EventRsvp.rsvpTime))
-    eventParticipantUsers = [participantDatum.user for participantDatum in eventParticipantData]
-    eventRsvpData = [rsvpDatum for rsvpDatum in eventRsvpData if rsvpDatum.user not in eventParticipantUsers]
+        # error handling depending on the boolean returned from updateEventParticipants
+        if volunteerUpdated:
+            flash("Volunteer table succesfully updated", "success")
+        else:
+            flash("Error adding volunteer", "danger")
+        return redirect(url_for("admin.manageVolunteersPage", eventID=eventID))
+    
+    # ------------ GET request ------------
+    elif request.method == "GET":
+        if not (g.current_user.isCeltsAdmin or (g.current_user.isCeltsStudentStaff and g.current_user.isProgramManagerForEvent(event))):
+            abort(403)
 
-    if event.isPast:
-        eventVolunteerData = eventParticipantData
-        eventNonAttendedData = eventRsvpData
-        eventWaitlistData = []
-    else:
-        eventWaitlistData = [volunteer for volunteer in eventParticipantData + eventRsvpData if volunteer.rsvpWaitlist and event.isRsvpRequired]
-        eventVolunteerData = [volunteer for volunteer in eventRsvpData if volunteer not in eventWaitlistData]
-        eventNonAttendedData = []
-        
-    program = event.program
+        # ------- Grab the different lists of participants -------
+        trainedParticipantsForProgramAndTerm = trainedParticipants(event.program, event.term)
 
-    allRelevantUsers = [participant.user for participant in eventVolunteerData + eventNonAttendedData + eventWaitlistData]
-    completedTrainingInfo = getParticipationStatusForTrainings(program, allRelevantUsers, event.term)
+        bannedUsersForProgram = [bannedUser.user for bannedUser in getBannedUsers(event.program)]
+  
+        eventNonAttendedData, eventWaitlistData, eventVolunteerData, eventParticipants = sortParticipantsByStatus(event)
+        
+        allRelevantUsers = [participant.user for participant in (eventParticipants + eventNonAttendedData + eventWaitlistData)]
```

Comment:
Does that handle duplicated users?

Reply:
There should never be a duplicate in any of those, or else something broke somewhere else. If necessary I could convert to a set and back, but I don't think that is necessary.